app/jobs/set_collection_job.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+##
+# job to move assign object to a new collection
+class SetCollectionJob < GenericJob
+  attr_reader :new_collection_ids
+
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  # @option params [Array] :pids required list of pids
+  # @option params [Hash] :set_collection
+  def perform(bulk_action_id, params)
+    super
+
+    @new_collection_ids = [params[:set_collection]['new_collection_id']]
+
+    with_bulk_action_log do |log|
+      log.puts("#{Time.current} Starting SetCollectionJob for BulkAction #{bulk_action_id}")
+      update_druid_count
+
+      pids.each do |current_druid|
+        log.puts("#{Time.current} SetCollectionJob: Starting update for #{current_druid} (bulk_action.id=#{bulk_action_id})")
+        set_collection_and_index_safely(current_druid, log)
+        log.puts("#{Time.current} SetGoverningApoJob: Finished update for #{current_druid} (bulk_action.id=#{bulk_action_id})")
+      end
+
+      log.puts("#{Time.current} Finished SetGoverningApoJob for BulkAction #{bulk_action_id}")
+    end
+  end
+
+  private
+
+  def set_collection_and_index_safely(current_druid, log)
+    cocina_item = Dor::Services::Client.object(current_druid).find
+    state_service = StateService.new(current_druid, version: cocina_item.version)
+    check_can_set_collection!(cocina_item, state_service)
+    open_new_version(current_druid, cocina_item.version, 'Set collection') unless state_service.allows_modification?
```

Comment:
This will require that you increment the version of the `cocina_item`, otherwise you will hit a version mismatch error when you save to DSA.  See the discussion on https://github.com/sul-dlss/argo/pull/3167#issuecomment-1059628010 and https://github.com/sul-dlss/argo/pull/3167/files#diff-039205bb74087606d756e6cba1c6c521615d03f1f7bebb6acecb1cce5ca3c9a6R42-R45 for the fix.

Reply:
@jcoyne done.